backend/src/main/java/com/capstone/moneytree/service/impl/DefaultUserService.java

```diff
@@ -371,6 +376,32 @@ public void deleteUserByEmail(String email) {
     }
 
     @Override
+
+    public List<User> getTopUsers(String symbol) {
+        List<User> users = new ArrayList<>();
+        // get a list of all user who own a stock with this symbol
+        ownsDao.findAll().stream()
+                .filter(owns -> owns.getStock().getSymbol().equals(symbol))
```

Comment:
Hi Razine, I tried this branch and realized that there are 2 bugs:
1) I know that right now, when we create a user we dedicate a score=0 to him/her, but for when we are using old users, they all have null score, so the compare method will throw null exception for when we are comparing the scores. Please change this line to this:
`  .filter(owns -> owns.getStock().getSymbol().equals(symbol) && owns.getUser().getScore()!=null)`
It is hard to make transactions that are updated inside alpaca market hours, so this would save us from resetting the database users to avoid null pointer.

2) I tried the endpoint with AAPL stock, with a user that had two AAPL transactions, I mean he bought AAPL in 2 different transactions. The endpoint returned the user 2 times, so I think we need to either use set<User> to avoid repetitions in our response.

Reply:
Good catch, will add the changes

Reply:
Hey Hossein, for the second points, even if the user made the transaction 2 times, he still owns the stock no? since I sort using the Owns relationship. If he owns the stock two times it is not good no ?

Reply:
I meant, we should return the list of 10 distinct users, but I got 1 user two times in the response.